sixtyfps_compiler/expression_tree.rs

```diff
@@ -398,6 +388,16 @@ pub enum Expression {
     EnumerationValue(EnumerationValue),
 
     ReturnStatement(Option<Box<Expression>>),
+
+    LayoutCacheAccess {
+        layout_cache_prop: NamedReference,
+        index: usize,
+        /// When set, this is the index within a repeater, and the index is then the location of another offset.
+        /// So this looks like `layout_cache_prop[layout_cache_prop[index] + repeater_index]`
+        repeater_index: Option<Box<Expression>>,
```

Comment:
Maybe it's worth expression this using an enum instead, since users have to check repeater_index anyway? Like

```rust
enum LayoutCacheIndex {
    ItemIndex(usize),
    ItemInRepeater {
        repeater_index: Box<Expression>,
        index_in_repeater: usize
    }
}
```

Not sure it's worth it though, just a though.

Reply:
Yeah, that would be nicer. but would force a different enum and more matching.  I can see that this structure will change in the future if we get smarter about how the cache is represented.

Reply:
Ok :)